service/src/intTest/java/uk/nhs/adaptors/gp2gp/gpc/GetGpcStructuredComponentTest.java

```diff
@@ -56,27 +62,36 @@ public void When_NewStructuredTask_Expect_DatabaseUpdatedAndAddedToObjectStore()
         var ehrExtractUpdated = ehrExtractStatusRepository.findByConversationId(ehrExtractStatus.getConversationId()).get();
         assertThatInitialRecordWasUpdated(ehrExtractUpdated, ehrExtractStatus);
 
-        var storageDataWrapper = getStorageDataWrapper(ehrExtractUpdated);
+        var storageDataWrapper = getStorageDataWrapper(ehrExtractUpdated.getGpcAccessStructured().getObjectName());
         assertThatObjectCreated(storageDataWrapper, ehrExtractUpdated, structuredTaskDefinition);
     }
 
     @Test
-    public void When_StructuredTaskRunTwice_Expect_ObjectToBeOverwrittenInStroage() throws IOException {
+    @SneakyThrows
+    public void When_StructuredTaskRunTwice_Expect_ObjectToBeOverwrittenInStroage() {
         var ehrExtractStatus = EhrExtractStatusTestUtils.prepareEhrExtractStatus();
         ehrExtractStatusRepository.save(ehrExtractStatus);
 
-        GetGpcStructuredTaskDefinition structuredTaskDefinition1 = buildValidStructuredTask(ehrExtractStatus);
-        getGpcStructuredTaskExecutor.execute(structuredTaskDefinition1);
+        GetGpcStructuredTaskDefinition structuredTaskDefinition = buildValidStructuredTask(ehrExtractStatus);
```

Comment:
Use the same exact task defintion (including same static taskId, see below). This is what would happen in a real system. The taskdef message is not ack'ed off the queue and is retried.